Sources/SourceKitLSP/Swift/DocumentFormatting.swift

```diff
@@ -139,26 +139,58 @@ private func edits(from original: DocumentSnapshot, to edited: String) -> [TextE
 extension SwiftLanguageService {
   package func documentFormatting(_ req: DocumentFormattingRequest) async throws -> [TextEdit]? {
     return try await format(
+      snapshot: documentManager.latestSnapshot(req.textDocument.uri),
       textDocument: req.textDocument,
       options: req.options
     )
   }
 
   package func documentRangeFormatting(_ req: DocumentRangeFormattingRequest) async throws -> [TextEdit]? {
     return try await format(
+      snapshot: documentManager.latestSnapshot(req.textDocument.uri),
       textDocument: req.textDocument,
       options: req.options,
       range: req.range
     )
   }
 
+  package func documentOnTypeFormatting(_ req: DocumentOnTypeFormattingRequest) async throws -> [TextEdit]? {
+    guard let server = self.sourceKitLSPServer else {
+      return nil
+    }
+    let snapshot = try documentManager.latestSnapshot(req.textDocument.uri)
+    let capabilities = await server.serverCapabilities(
+      for: capabilityRegistry.clientCapabilities,
+      registry: capabilityRegistry
+    )
+    
+    guard let documentOnTypeFormattingProvider = capabilities.documentOnTypeFormattingProvider,
+      documentOnTypeFormattingProvider.firstTriggerCharacter == req.ch
+        || documentOnTypeFormattingProvider.moreTriggerCharacter?.contains(req.ch) == true,
```

Comment:
I generally find `?? false` easier to read because it directly answers the question: What happens if this is `nil`?

```suggestion
        || (documentOnTypeFormattingProvider.moreTriggerCharacter?.contains(req.ch) ?? false),
```

Reply:
I personally find `== true` more understandable, but I've seen some other people like `?? false` instead.